src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -2069,6 +2115,418 @@ void CalcParallelPIU(EnergyPlusData &state,
+        } else if (coolingMode != CoolOpModeType::CoolerOff && heaterMode == HeatOpModeType::HeaterOff) {
+            state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = constantVolumeCooling;
+        } else {
+            state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = deadband;
+        }
+    }
+    if (state.dataPowerInductionUnits->PIU(PIUNum).fanControlType == FanCntrlType::VariableSpeedFan) {
+        if (state.dataPowerInductionUnits->PIU(PIUNum).heatingControlType == HeatCntrlBehaviorType::StagedHeaterBehavior) {
+            if (heaterMode == HeatOpModeType::StagedHeatFirstStage) {
+                state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedStagedHeatFirstStageHeating;
+            } else if (heaterMode == HeatOpModeType::StagedHeatSecondStage) {
+                state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedStagedHeatSecondStageHeating;
+            }
+        } else if (state.dataPowerInductionUnits->PIU(PIUNum).heatingControlType == HeatCntrlBehaviorType::ModulatedHeaterBehavior) {
+            if (heaterMode == HeatOpModeType::ModulatedHeatFirstStage) {
+                state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedModulatedHeatFirstStageHeating;
+            } else if (heaterMode == HeatOpModeType::ModulatedHeatSecondStage) {
+                state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedModulatedHeatSecondStageHeating;
+            } else if (heaterMode == HeatOpModeType::ModulatedHeatThirdStage) {
+                state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedModulatedHeatThirdStageHeating;
+            }
+        }
+        if (coolingMode == CoolOpModeType::CoolFirstStage) {
+            state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedFirstStageCooling;
+        }
+        if (coolingMode == CoolOpModeType::CoolSecondStage) {
+            state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = variableSpeedSecondStageCooling;
+        }
+        if (heaterMode == HeatOpModeType::HeaterOff && coolingMode == CoolOpModeType::CoolerOff) {
+            state.dataPowerInductionUnits->PIU(PIUNum).CurOperationControlStage = deadband;
+        }
+    }
+}
+
+void CalcVariableSpeedPIUCoolingBehavior(EnergyPlusData &state,
+                                         int const PIUNum,   // number of the current PIU being simulated
+                                         int const ZoneNode, // zone node number
+                                         Real64 const zoneLoad,
+                                         Real64 const loadToHeatSetPt,
+                                         bool const Pri,
+                                         Real64 const PriAirMassFlowMin,
+                                         Real64 const PriAirMassFlowMax)
+{
```

Comment:
New function used the determine if the variable speed fan should run at minimum, maximum, or in between the minimum and maximum to meet the load. If the latter, use regula falsi to find at which fraction it needs to run to meet the load.